src/util/projection.ts

```diff
@@ -0,0 +1,51 @@
+import {ProjectionDefinitionT, ProjectionDefinitionSpecification, PropertyValueSpecification} from '../types.g';
+
+export default class ProjectionDefinition extends Array {
+    constructor(from: string, to: string, transition: number){
+        super(3);
+        this[0] = from;
+        this[1] = to;
+        this[2] = transition;
+    }
+    
+    get from() { return this[0]; }
+    get to() { return this[1]; }
+    get transition() { return this[2]; }
+
+    toString() {
+        return `["${this[0]}", "${this[1]}", ${this[2]}]`;
+    }
+
+    static parse(input: any): ProjectionDefinition {
+        if (input instanceof ProjectionDefinition) {
+            return input;
+        }
+        if (Array.isArray(input) && input.length === 3) {
+            return new ProjectionDefinition(input[0], input[1], input[2]);
+        }
+        if (typeof input === 'string') {
+            return new ProjectionDefinition(input, input, 1);
+        }
+        return undefined;
+    }
+}
+
+export function isProjectionDefinition(value: unknown): value is ProjectionDefinitionSpecification {
```

Comment:
Are all these exported methods still needed?

Reply:
managed to remove some. think the rest are